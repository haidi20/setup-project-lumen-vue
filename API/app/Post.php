<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Post extends Model
{
    public function searchAbleColumn()
    {
        return ['title', 'author', 'date'];
    }

    public function scopeSearch($query)
    {
        foreach ($this->searchAbleColumn() as $item) {
            $query->orWhere($item, 'like', '%' . request('search') . '%');
        }
    }
}
