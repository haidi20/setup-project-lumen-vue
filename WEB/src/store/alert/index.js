// partials
import alertStates from './_partials/alertStates'
import alertGetters from './_partials/alertGetters'
import alertMutations from './_partials/alertMutations'
import alertActions from './_partials/alertActions'

const alert = {
    namespaced: true,
    state: alertStates,
    getters: alertGetters,
    actions: alertActions,
    mutations: alertMutations,
}

export default alert