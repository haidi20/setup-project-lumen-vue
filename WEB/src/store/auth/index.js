import authStates from './authStates';
import authGetters from './authGetters';
import authActions from './authActions';
import authMutations from './authMutations';

const auth = {
    namespaced: true,
    state: authStates,
    getters: authGetters,
    actions: authActions,
    mutations: authMutations,
}

export default auth