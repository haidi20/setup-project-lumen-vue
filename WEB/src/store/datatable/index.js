import http from '../../api'

const datatable = {
    namespaced: true,
    state: {
        data: [],
        headers: [
            { text: "No.", sortable: false, value: "number", class: "bg-header-info" },
            { text: "Actions", value: "actions", sortable: false, align: "start", class: "bg-header-info" },
            { text: "Title", sortable: false, value: "title", class: "bg-header-info" },
            { text: "Date", value: "date", sortable: false, class: "bg-header-info" },
            { text: "Author", value: "author", sortable: false, class: "bg-header-info" },
        ],
        page: 1,
        totalPages: 0,
        pageSize: 5,
        pageSizes: [5, 10],
        loading: false,
        search: '',
        awaitingSearch: false,
    },
    getters: {
        getNumberData: state => item => {
            let number = state.data.map(function(x) { return x.id; }).indexOf(item.id) + 1;

            if (state.page == 1) {
                return number;
            } else {
                return (number + state.pageSize);
            }
        },
    },
    mutations: {
        NULL_DATA: (state) => {
            state.data = [];
        },
        INSERT_DATA: (state, payload) => {
            state.data = payload.data;
            state.totalPages = payload.last_page;
        },
        LOADING_TRUE: (state) => {
            state.loading = true;
        },
        LOADING_FALSE: (state) => {
            state.loading = false;
        },
    },
    actions: {
        async fetchData({ commit }, payload) {
            // const params = {
            //     page: state.page,
            //     search: state.search,
            //     per_page: state.pageSize,
            // }
            commit('NULL_DATA');
            commit('LOADING_TRUE');
            try {
                await http.get(`/api${payload.dataLink}`)
                    .then(ress => {
                        let data = ress.data.data;
                        commit('INSERT_DATA', data);
                        commit('LOADING_FALSE');
                    });
            } catch (error) {
                console.log('error fetch data = ' + error);
                return (null);
            }
        },
        // PageChange({ commit }) {
        //     // state.page = value;
        //     // this.fetchData();
        // },
        // PageSizeChange({ commit }) {
        //     // state.page = 1;
        //     // state.pageSize = size;
        //     // state.fetchData();
        // },
        // watchSearch: function({ commit }) {
        //     // if (!state.awaitingSearch) {
        //     //     setTimeout(() => {
        //     //         // this.fetchData();
        //     //         state.awaitingSearch = false;
        //     //     }, 2000); // 2 sec delay
        //     // }

        //     // state.awaitingSearch = true;
        // },
        // save_donasi({ commit, rootState }, payload) {
        //     // rootState BERARTI MENGAKSES STATE YANG TIDAK BERADA DALAM MODULES
        //     // DALAM HAL INI STATE isLoading YANG ADA DI DALAM FILE store.js
        //     rootState.isLoading = true //SET TRUE UNTUK MEMBERIKAN EFEK LOADING
        //     setTimeout(() => {
        //         //MENGINSTRUKSIKAN PADA MUTATIONS TERKAIT UNTUK MENJALANKAN INSTRUKSINYA
        //         commit('ADD_DONASI', payload)
        //         // STATE isLoading DI MATIKAN KEMBALI
        //         rootState.isLoading = false
        //     }, 1000)
        // }
    }
}

export default datatable