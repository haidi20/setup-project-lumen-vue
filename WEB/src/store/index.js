import Vue from 'vue'
import Vuex from 'vuex'

import auth from './auth'
import alert from './alert'
import datatable from './datatable'

Vue.use(Vuex)

export default new Vuex.Store({
    modules: {
        auth,
        alert,
        datatable,
    }
});