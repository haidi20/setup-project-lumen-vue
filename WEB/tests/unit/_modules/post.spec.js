import Vue from 'vue';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

import { mount, shallowMount, createLocalVue } from '@vue/test-utils';

//components
import Post from '@/components/post';

import store from '@/store';

Vue.use(Vuex);
Vue.use(Vuetify);

describe('Post', () => {
    const localVue = createLocalVue()
    let vuetify, wrapperMount, wrapperShallowMount;

    beforeEach(() => {
        vuetify = new Vuetify();
        wrapperMount = mount(Post, {
            localVue,
            vuetify,
            store,
        });
        wrapperShallowMount = shallowMount(Post, {
            localVue,
            vuetify,
            store,
        });
    });

    it('check word title', async() => {
        const title = wrapperMount.find('h2');

        expect(title.text()).toBe('Posts');
    });

    it('check data count 5', async() => {
        const data = await wrapperMount.findAll('.v-data-table__wrapper > table > tr');
        expect(data.length).toBe(5);
    });

    it('check default value PageSizeChange', () => {
        const select = wrapperMount.find('.v-select').props('value');
        expect(select).toBe(5);
    });

    it('event click pageSizeChange value 10', async() => {
        await wrapperMount.find('.v-select__slot').trigger('click');
        await wrapperMount.findAll('.v-list-item').at(1).trigger('click');
        const value = wrapperMount.find('.v-select__selection.v-select__selection--comma');
        expect(value.text()).toBe("10");
    });
});