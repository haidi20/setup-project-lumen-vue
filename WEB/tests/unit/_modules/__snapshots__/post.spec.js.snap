// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post event click pageSizeChange 1`] = `
<div class="container container--fluid">
  <div xs="12" class="col-sm-12 col-md-12 col">
    <h2>Posts</h2>
    <div>
      <div class="space-top-datatable">
        <p style="display: inline-block;">show</p>
        <div class="v-input select-page-size v-input--is-label-active v-input--is-dirty v-input--is-focused theme--light v-text-field v-select v-select--is-menu-active primary--text">
          <div class="v-input__control">
            <div role="button" aria-haspopup="listbox" aria-expanded="true" aria-owns="list-45" class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">10</div><input id="pageSize" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light primary--text"></i></div>
                </div><input type="hidden" value="10">
              </div>
              <div class="v-menu">
                <transition-stub name="v-menu-transition">
                  <div class="v-menu__content theme--light menuable__content__active " style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 8; display: none;">
                    <div role="listbox" tabindex="-1" class="v-list v-select-list v-sheet theme--light theme--light" id="list-45">
                      <div tabindex="0" aria-selected="false" id="list-item-64-0" role="option" class="v-list-item v-list-item--link theme--light">
                        <div class="v-list-item__content">
                          <div class="v-list-item__title">5</div>
                        </div>
                      </div>
                      <div tabindex="0" aria-selected="true" id="list-item-64-1" role="option" class="v-list-item primary--text v-list-item--active v-list-item--link theme--light">
                        <div class="v-list-item__content">
                          <div class="v-list-item__title">10</div>
                        </div>
                      </div>
                      <div tabindex="0" aria-selected="false" id="list-item-64-2" role="option" class="v-list-item v-list-item--link theme--light">
                        <div class="v-list-item__content">
                          <div class="v-list-item__title">20</div>
                        </div>
                      </div>
                    </div>
                  </div>
                </transition-stub>
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light primary--text">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
        <p style="display: inherit;">entires</p>
        <div class="space-search">
          <div class="v-input search theme--light v-text-field v-text-field--single-line">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-text-field__slot"><label for="input-50" class="v-label theme--light" style="left: 0px; position: absolute;">search</label><input id="input-50" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-table elevation-1 theme--light">
        <div class="v-data-table__wrapper">
          <table>
            <colgroup>
              <col class="">
              <col class="">
              <col class="">
              <col class="">
              <col class="">
            </colgroup>
            <thead class="v-data-table-header">
              <tr>
                <th role="columnheader" scope="col" aria-label="No." class="text-start"><span>No.</span></th>
                <th role="columnheader" scope="col" aria-label="Actions" class="text-start"><span>Actions</span></th>
                <th role="columnheader" scope="col" aria-label="Title" class="text-start"><span>Title</span></th>
                <th role="columnheader" scope="col" aria-label="Date" class="text-start"><span>Date</span></th>
                <th role="columnheader" scope="col" aria-label="Author" class="text-start"><span>Author</span></th>
              </tr>
            </thead>
            <thead>
              <tr class="v-data-table__progress">
                <th colspan="5" class="column">
                  <div role="progressbar" aria-valuemin="0" aria-valuemax="100" class="v-progress-linear v-progress-linear--absolute theme--light" style="height: 4px;">
                    <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 0%; width: 100%;"></div>
                    <div class="v-progress-linear__buffer"></div>
                    <transition-stub name="fade-transition">
                      <div class="v-progress-linear__indeterminate v-progress-linear__indeterminate--active">
                        <div class="v-progress-linear__indeterminate long primary"></div>
                        <div class="v-progress-linear__indeterminate short primary"></div>
                      </div>
                    </transition-stub>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr class="v-data-table__empty-wrapper">
                <td colspan="5">Loading..</td>
              </tr>
            </tbody>
          </table>
        </div>
      </div> <br>
      <nav role="navigation" aria-label="Pagination Navigation" style="float: right;">
        <ul class="v-pagination v-pagination--disabled theme--light">
          <li><button type="button" aria-label="Previous page" class="v-pagination__navigation v-pagination__navigation--disabled"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-left theme--light"></i></button></li>
          <li><button type="button" aria-label="Next page" class="v-pagination__navigation v-pagination__navigation--disabled"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-right theme--light"></i></button></li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;

exports[`Post event click pageSizeChange value 10 1`] = `
<div class="container container--fluid">
  <div xs="12" class="col-sm-12 col-md-12 col">
    <h2>Posts</h2>
    <div>
      <div class="space-top-datatable">
        <p style="display: inline-block;">show</p>
        <div class="v-input select-page-size v-input--is-label-active v-input--is-dirty v-input--is-focused theme--light v-text-field v-select v-select--is-menu-active primary--text">
          <div class="v-input__control">
            <div role="button" aria-haspopup="listbox" aria-expanded="true" aria-owns="list-65" class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">10</div><input id="pageSize" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light primary--text"></i></div>
                </div><input type="hidden" value="10">
              </div>
              <div class="v-menu">
                <transition-stub name="v-menu-transition">
                  <div class="v-menu__content theme--light menuable__content__active " style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 8; display: none;">
                    <div role="listbox" tabindex="-1" class="v-list v-select-list v-sheet theme--light theme--light" id="list-65">
                      <div tabindex="0" aria-selected="false" id="list-item-84-0" role="option" class="v-list-item v-list-item--link theme--light">
                        <div class="v-list-item__content">
                          <div class="v-list-item__title">5</div>
                        </div>
                      </div>
                      <div tabindex="0" aria-selected="true" id="list-item-84-1" role="option" class="v-list-item primary--text v-list-item--active v-list-item--link theme--light">
                        <div class="v-list-item__content">
                          <div class="v-list-item__title">10</div>
                        </div>
                      </div>
                      <div tabindex="0" aria-selected="false" id="list-item-84-2" role="option" class="v-list-item v-list-item--link theme--light">
                        <div class="v-list-item__content">
                          <div class="v-list-item__title">20</div>
                        </div>
                      </div>
                    </div>
                  </div>
                </transition-stub>
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light primary--text">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
        <p style="display: inherit;">entires</p>
        <div class="space-search">
          <div class="v-input search theme--light v-text-field v-text-field--single-line">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-text-field__slot"><label for="input-70" class="v-label theme--light" style="left: 0px; position: absolute;">search</label><input id="input-70" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-table elevation-1 theme--light">
        <div class="v-data-table__wrapper">
          <table>
            <colgroup>
              <col class="">
              <col class="">
              <col class="">
              <col class="">
              <col class="">
            </colgroup>
            <thead class="v-data-table-header">
              <tr>
                <th role="columnheader" scope="col" aria-label="No." class="text-start"><span>No.</span></th>
                <th role="columnheader" scope="col" aria-label="Actions" class="text-start"><span>Actions</span></th>
                <th role="columnheader" scope="col" aria-label="Title" class="text-start"><span>Title</span></th>
                <th role="columnheader" scope="col" aria-label="Date" class="text-start"><span>Date</span></th>
                <th role="columnheader" scope="col" aria-label="Author" class="text-start"><span>Author</span></th>
              </tr>
            </thead>
            <thead>
              <tr class="v-data-table__progress">
                <th colspan="5" class="column">
                  <div role="progressbar" aria-valuemin="0" aria-valuemax="100" class="v-progress-linear v-progress-linear--absolute theme--light" style="height: 4px;">
                    <div class="v-progress-linear__background primary" style="opacity: 0.3; left: 0%; width: 100%;"></div>
                    <div class="v-progress-linear__buffer"></div>
                    <transition-stub name="fade-transition">
                      <div class="v-progress-linear__indeterminate v-progress-linear__indeterminate--active">
                        <div class="v-progress-linear__indeterminate long primary"></div>
                        <div class="v-progress-linear__indeterminate short primary"></div>
                      </div>
                    </transition-stub>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr class="v-data-table__empty-wrapper">
                <td colspan="5">Loading..</td>
              </tr>
            </tbody>
          </table>
        </div>
      </div> <br>
      <nav role="navigation" aria-label="Pagination Navigation" style="float: right;">
        <ul class="v-pagination v-pagination--disabled theme--light">
          <li><button type="button" aria-label="Previous page" class="v-pagination__navigation v-pagination__navigation--disabled"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-left theme--light"></i></button></li>
          <li><button type="button" aria-label="Next page" class="v-pagination__navigation v-pagination__navigation--disabled"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-right theme--light"></i></button></li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
